### PR TITLE
Feature/update authorize role

### PR DESCRIPTION
**Pull request by poonammohata**
Resolves - #615

- replaced authorizeUsers middleware with authorizeRoles middleware
- removed LEGACY_ROLES from 'constants.roles.js' file
- where Roles are passed as string in authorize middleware updated that to Role's key

### CODE REVIEW

File: constants/roles.js [hunk 1]
```diff
@@ -1,10 +1,4 @@
-// Use Legacy Roles with authorizeUsers middleware
-const LEGACY_ROLES = {
-  SUPER_USER: "superUser",
-  APP_OWNER: "appOwner",
-};
-
```
Comment (pallabez) [inline constants/roles.js:6]: Thanks for removing this as well.

File: routes/tasks.js [hunk 2]
```diff
@@ -159,7 +162,7 @@ router.post("/", authenticate, authorizeUser("appOwner"), createTask, tasks.addN
  *           schema:
  *             $ref: '#/components/schemas/errors/badImplementation'
  */
-router.patch("/:id", authenticate, authorizeUser("appOwner"), updateTask, tasks.updateTask);
+router.patch("/:id", authenticate, authorizeRoles([APPOWNER]), updateTask, tasks.updateTask);
```
Comment (pallabez) [inline routes/tasks.js:165]: Previous middleware `authorizeUser("appOwner")` gives permission to both app_owener and super_user access. 
The current only gives it to mentioned role. So can we add SUPERUSER to the array as well? 
```suggestion
router.patch("/:id", authenticate, authorizeRoles([APPOWNER, SUPERUSER]), updateTask, tasks.updateTask);
```

Comment (poonammohata) [reply routes/tasks.js:165]: okh that means where ever APPOWNER role is given, we need to pass SUPERUSER role also in that array, am i understanding it right? @pallabez

**Review by pallabez — DISMISSED** [review]
LGTM :+1:

File: routes/tasks.js [hunk 3]
```diff
@@ -3,13 +3,16 @@ const router = express.Router();
...
+} = require("../constants/roles");
 
 router.get("/", tasks.fetchTasks);
 router.get("/self", authenticate, tasks.getSelfTasks);
-router.get("/overdue", authenticate, authorizeUser("superUser"), tasks.overdueTasks);
-router.post("/", authenticate, authorizeUser("appOwner"), createTask, tasks.addNewTask);
-router.patch("/:id", authenticate, authorizeUser("appOwner"), updateTask, tasks.updateTask);
+router.get("/overdue", authenticate, authorizeRoles([SUPERUSER]), tasks.overdueTasks);
+router.post("/", authenticate, authorizeRoles([APPOWNER]), createTask, tasks.addNewTask);
+router.patch("/:id", authenticate, authorizeRoles([APPOWNER]), updateTask, tasks.updateTask);
```
Comment (pallabez) [inline routes/tasks.js:15]:
```suggestion
router.post("/", authenticate, authorizeRoles([APPOWNER, SUPERUSER]), createTask, tasks.addNewTask);
router.patch("/:id", authenticate, authorizeRoles([APPOWNER, SUPERUSER]), updateTask, tasks.updateTask);
```

Previously `appOwner` gave permission to both app_owner and super_user. 
Now it gives access to only the role mentioned.

File: constants/roles.js [hunk 4]
```diff
@@ -1,10 +1,4 @@
...
 // Use Roles with authorizeRoles middleware
 const ROLES = { SUPERUSER: "super_user", APPOWNER: "app_owner", MEMBER: "member", ARCHIVED: "archived" };
```
Comment (akshay1502) [inline constants/roles.js:2]: Can we have each key value pair on a new line.

Comment (poonammohata) [reply constants/roles.js]: made the changes

**Review by pallabez — APPROVED** [review]
Lgtm

**Review by pallabez — APPROVED** [review]
Thanks for the changes and keeping track of it

**Pull request merged**